ng-dev/utils/git/authenticated-git-client.ts

```diff
@@ -81,6 +85,30 @@ export class AuthenticatedGitClient extends GitClient {
     return {error};
   }
 
+  /**
+   * Gets an owned fork for the configured project of the authenticated user, caching the determined
+   * fork repository as the authenticated user cannot change during action execution.
+   */
+  async getForkOfAuthenticatedUser(): Promise<GithubRepo> {
+    if (this._cachedForkRepo !== null) {
+      return this._cachedForkRepo;
+    }
+
+    const {owner, name} = this.remoteConfig;
+    const result = await this.github.graphql(findOwnedForksOfRepoQuery, {owner, name});
+    const forks = result.repository.forks.nodes;
+
+    if (forks.length === 0) {
+      throw Error(
+        'Unable to find fork for currently authenticated user. Please ensure you created a fork ' +
```

Comment:
We should rather return `null` as this is much easier to check and less ambiguous. Especially since we throw a non-unique error here it is difficult to filter out if some logic just failed by mistake vs. if just a fork could not be found.